specs/Qowaiv.Specs/Chemistry/CAS_Registry_Number_specs.cs

```diff
@@ -0,0 +1,489 @@
+﻿using Qowaiv.Chemistry;
+
+namespace Chemistry.CAS_Registry_Number_specs;
+
+public class With_domain_logic
+{
+    [TestCase("")]
+    [TestCase("?")]
+    public void has_length_zero_for_empty_and_unknown(CasRegistryNumber svo)
+        => svo.Length.Should().Be(0);
+
+    
+    [TestCase(5, "73–24–5")]
+    [TestCase(7, "7732-18-5")]
+    [TestCase(8, "10028-14-5")]
+    public void has_length(int length, CasRegistryNumber svo)
+        => svo.Length.Should().Be(length);
+
+    [TestCase(false, "10028-14-5")]
+    [TestCase(false, "?")]
+    [TestCase(true, "")]
+    public void IsEmpty_returns(bool result, CasRegistryNumber svo)
+        => svo.IsEmpty().Should().Be(result);
+
+    [TestCase(false, "10028-14-5")]
+    [TestCase(true, "?")]
+    [TestCase(true, "")]
+    public void IsEmptyOrUnknown_returns(bool result, CasRegistryNumber svo)
+        => svo.IsEmptyOrUnknown().Should().Be(result);
+
+    [TestCase(false, "10028-14-5")]
+    [TestCase(true, "?")]
+    [TestCase(false, "")]
+    public void IsUnknown_returns(bool result, CasRegistryNumber svo)
+        => svo.IsUnknown().Should().Be(result);
+}
+
+public class Is_valid_for
+{
+    [TestCase("?")]
+    [TestCase("unknown")]
+    public void strings_representing_unknown(string input)
+        => CasRegistryNumber.IsValid(input).Should().BeTrue();
+
+    [TestCase("10028-14-5", "nl")]
+    [TestCase("10028-14-5", "nl")]
+    public void strings_representing_SVO(string input, CultureInfo culture)
+        => CasRegistryNumber.IsValid(input, culture).Should().BeTrue();
+}
+
+public class Is_not_valid_for
+{
+    [Test]
+    public void Numbers_with_less_then_5_digits()
+        => CasRegistryNumber.IsValid("9-99-4").Should().BeFalse();
+
+    [Test]
+    public void Numbers_with_more_then_10_digits()
+        => CasRegistryNumber.IsValid("10000000-00-0").Should().BeFalse();
+
+    [Test]
+    public void string_empty()
+        => CasRegistryNumber.IsValid(string.Empty).Should().BeFalse();
+
+    [Test]
+    public void string_null()
+        => CasRegistryNumber.IsValid(null).Should().BeFalse();
+
+    [Test]
+    public void whitespace()
+        => CasRegistryNumber.IsValid(" ").Should().BeFalse();
+
+    [TestCase]
+    public void garbage()
+        => CasRegistryNumber.IsValid("garbage").Should().BeFalse();
+
+    [TestCase("10028-14-3")]
+    [TestCase("10028-15-5")]
+    [TestCase("10028-84-5")]
+    [TestCase("10020-14-5")]
+    [TestCase("10078-14-5")]
+    [TestCase("10128-14-5")]
+    [TestCase("32028-14-5")]
+    public void checksum_mismatches(string number)
```

Comment:
Should there be testcases for cas numbers with different lengths?

Reply:
The point here is that by changing one digit, the valid number was not longer valid.